sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeArrayData.java

```diff
@@ -463,23 +463,35 @@ public static UnsafeArrayData fromPrimitiveArray(
     final long[] data = new long[(int)totalSizeInLongs];
 
     Platform.putLong(data, Platform.LONG_ARRAY_OFFSET, length);
-    if (arr != null) {
```

Comment:
shall we keep this if?

Reply:
Originally, this if is not here. It is fine with me to keep this for the safety.